cli/ballerina-cli/src/main/java/io/ballerina/cli/launcher/CustomToolClassLoader.java

```diff
@@ -28,23 +33,71 @@
  * @since 2201.8.0
  */
 public class CustomToolClassLoader extends URLClassLoader {
+    private final ClassLoader system;
 
     public CustomToolClassLoader(URL[] urls, ClassLoader parent) {
         super(urls, parent);
+        system = getSystemClassLoader();
     }
 
     @Override
     protected Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
-        try {
-            // Load from parent if cli or picocli classes. This is to avoid SPI and class loading issues
-            if (name.startsWith("io.ballerina.cli") || name.startsWith("picocli")) {
-                return super.loadClass(name, resolve);
+        Class<?> loadedClass = findLoadedClass(name);
+        if (loadedClass == null) {
+            try {
+                // First, try to load the class from the URLs
+                loadedClass = findClass(name);
+            } catch (ClassNotFoundException e) {
+                try {
+                    // If not found, delegate to the parent
+                    loadedClass = super.loadClass(name, resolve);
+                } catch (ClassNotFoundException e2) {
+                    // If not found, delegate to the system class loader
+                    if (system != null) {
+                        loadedClass = system.loadClass(name);
+                    }
+                }
             }
-            // First, try to load the class from the URLs
-            return findClass(name);
-        } catch (ClassNotFoundException e) {
-            // If not found, delegate to the parent
-            return super.loadClass(name, resolve);
         }
+        if (resolve) {
+            resolveClass(loadedClass);
```

Comment:
This can be improved as `resolveClass(Objects.requireNonNull(loadedClass));`

Reply:
Fixed with ecdf19e8976b56ce29e9251d8f9d88351507e46a